### PR TITLE
Feat: Add uuid support

### PR DESCRIPTION
**Pull request by Leon0402**
Support for the uuid format of the json schema standard. Also added some more documentation to the readme.

### CODE REVIEW

**Comment by Leon0402** [comment]
Also added some tests

File: test/JSON-Schema-Test-Suite/CMakeLists.txt [hunk 1]
```diff
@@ -64,6 +64,7 @@ if(JSON_SCHEMA_TEST_SUITE_PATH)
             JSON-Suite::Optional::Format::uri-reference
             JSON-Suite::Optional::Format::uri-template
             JSON-Suite::Optional::Format::uri
+            JSON-Suite::Optional::Format::uuid
```
Comment (pboettch) [inline test/JSON-Schema-Test-Suite/CMakeLists.txt:68]: The test should not be added here: these are the tests expected to fail. `WILL_FAIL ON`.

Comment (Leon0402) [reply test/JSON-Schema-Test-Suite/CMakeLists.txt:68]: I wasn't quite sure where to add the tests. Can you provide some instructions?

Comment (pboettch) [reply test/JSON-Schema-Test-Suite/CMakeLists.txt:68]: Nowhere, it is found by a GLOB-call from CMake.

File: test/JSON-Schema-Test-Suite/tests/draft7/optional/format/uuid.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                "valid": true
+            },
+            {
+                "description": "an invalid uuid with hyphens at wrong position",
+                "data": "3fa85f645-717-4562b-3fc2c9-63f66afa6",
+                "valid": true
+            }
+        ]
+    }
+]
```
Comment (pboettch) [inline test/JSON-Schema-Test-Suite/tests/draft7/optional/format/uuid.json:30]: Where did you get this file from? Did you write it yourself? If so, maybe you should also send it to the JSON-Schema-Test-Suite-project.

Is `uuid` an officially mentioned/supported format-definition by the standard?

Comment (Leon0402) [reply test/JSON-Schema-Test-Suite/tests/draft7/optional/format/uuid.json:30]: Yes it is officially supported, in a newer standard than this lib currently supports though: 
https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.5

**Comment by Leon0402** [comment]
I wrote the test myself, because I thought that's what you did as well. 

UUID is actually tested by the project you mentioned: https://github.com/json-schema-org/JSON-Schema-Test-Suite/blob/master/tests/draft-future/optional/format/uuid.json. 

So I guess I can remove the test?

**Comment by pboettch** [comment]
Great, OK, everything is clear now.

Let's take (and copy) the test from 2020-12 to this project (this is how I did for the others tests, except I took the draft-7 ones) and integrate your changes.

**Comment by pboettch** [comment]
I integrated the changes related to the UUID-format-string-checker. Everything else was removed. Especially the changes to the CMakeLists.txt files. There is another PR open which handles this.

**Comment by Leon0402** [comment]
Oh no, the cmake changes were not supposed to be pushed anyway, I'm sorry. I just added / commented out some stuff so testing would be easy for me. 
It seems I pushed them yesterday as well.

**Pull request closed without merging**